packages/web3-utils/src/chunk_response_parser.ts

```diff
@@ -0,0 +1,86 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { JsonRpcResponse } from 'web3-types';
+import { InvalidResponseError } from 'web3-errors';
+
+export class ChunkResponseParser {
```

Comment:
Created https://github.com/ChainSafe/web3.js/issues/5349 for more tests we can do in another PR.